server/session/tournamentSession.js

```diff
@@ -122,27 +122,28 @@ const TournamentSession = {
             this.pairings = this.pairings.concat(pairings);
         }
 
-        return this.startPairingSessions().then(() => {
-            this.ranking.updateRatings();
-            this.ranking.players.forEach(ranking => {
-                this.rankingTable.updatePlayerRating(ranking.name, ranking.player.getRating());
+        return this.startPairingSessions()
+            .then(() => {
+                this.ranking.updateRatings();
+                this.ranking.players.forEach(ranking => {
+                    this.rankingTable.updatePlayerRating(ranking.name, ranking.player.getRating());
+                });
+                this.rankingTable.updateAndSortRanking();
+                this.clientApi.broadcastTournamentRankingTable(this.rankingTable);
             });
-            this.rankingTable.updateAndSortRanking();
-            this.clientApi.broadcastTournamentRankingTable(this.rankingTable);
-        });
     },
 
     rankPairing(pairing, result) {
-        let {player1, player2} = pairing;
+        let { player1, player2 } = pairing;
 
-        _.remove(this.pairings, pairing);
+        this.pairings.splice(this.pairings.findIndex(actPairing => actPairing === pairing), 1);
```

Comment:
So this is actually the effective change I had to do for making it work. _.remove will remove all matching elements. With findIndex I'll find only the first matching element and with splice I'll remove one element at the given Index of the pairings array...